src/lib.rs

```diff
@@ -335,4 +390,24 @@ mod tests {
         let cloned = host_entry.clone();
         assert_eq!(host_entry, cloned)
     }
+
+    #[test]
+    fn test_parse_hostfile() {
```

Comment:
This test seems to depend on Github Action's hostile specifically right? Ideally I'd like for these tests to work on any environment, but I can see the value of parsing the file for real (and validating that we choose the right path on windows vs Linux). 

Can we add an explicit comment at the top of this test saying that this test assumes that it's being run on github actions?

Reply:
Okay, now that part is only executed on GHA.

Note that still assumes that there is one entry but the default hostfile in Windows is empty. Do you wan't to remove that assumption too?

Reply:
Let's do that instead...

Reply:
Should be fine now.